ecs-cluster-infrastructure-service-build-pipeline-s3-artifact-store.tf

```diff
@@ -0,0 +1,111 @@
+resource "aws_s3_bucket" "infrastructure_ecs_cluster_service_build_pipeline_artifact_store" {
+  count = length(local.infrastructure_ecs_cluster_services) != 0 ? 1 : 0
+
+  bucket = "${local.resource_prefix_hash}-ecs-cluster-service-build-pipeline-artifact-store"
+
+}
+
+resource "aws_s3_bucket_policy" "infrastructure_ecs_cluster_service_build_pipeline_artifact_store" {
+  count = length(local.infrastructure_ecs_cluster_services) != 0 ? 1 : 0
+
+  bucket = aws_s3_bucket.infrastructure_ecs_cluster_service_build_pipeline_artifact_store[0].id
+  policy = templatefile(
+    "${path.module}/policies/s3-bucket-policy.json.tpl",
+    {
+      statement = <<EOT
+      [
+      ${templatefile("${path.root}/policies/s3-bucket-policy-statements/enforce-tls.json.tpl",
+      {
+        bucket_arn = aws_s3_bucket.infrastructure_ecs_cluster_service_build_pipeline_artifact_store[0].arn
+      }
+      )}${local.infrastructure_vpc_flow_logs_cloudwatch_logs && local.infrastructure_kms_encryption ? "," : ""}
+      ${templatefile("${path.root}/policies/s3-bucket-policy-statements/enforce-kms-encryption.json.tpl",
+      {
+        bucket_arn = local.infrastructure_kms_encryption ? aws_s3_bucket.infrastructure_ecs_cluster_service_build_pipeline_artifact_store[0].arn : ""
+      }
+  )}
+      ]
+      EOT
+}
```

Comment:
Is it possible to tidy up the indentation here? It's quite hard to scan read. Is it a big enough templatefile usage to justify a terraform template for this?

Reply:
The reason for it being formatted like this, is because the S3 policy is built up from multiple reusable conditional statements
The indentation issue is because of `terraform fmt`